SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorModule.cpp

```diff
@@ -82,6 +85,67 @@ bool FSpatialGDKEditorModule::CanExecuteLaunch() const
 	return SpatialGDKEditorInstance->GetPackageAssemblyRef()->CanBuild();
 }
 
+bool FSpatialGDKEditorModule::CanStartSession() const
+{
+	if (!SpatialGDKEditorInstance->IsSchemaGenerated())
+	{
+		EAppReturnType::Type Result = FMessageDialog::Open(EAppMsgType::YesNo, LOCTEXT("MissingSchema", "Attempted to start a local deployment but schema is not generated. Do you want to generate it?"));
+		if (Result == EAppReturnType::Yes)
+		{
+			// TODO generate schema
+		}
+
+		return false;
+	}
+
+	if (ShouldConnectToCloudDeployment())
+	{
+		if (GetDevAuthToken().IsEmpty())
+		{
+			FMessageDialog::Open(EAppMsgType::Ok, LOCTEXT("MissingDevelopmentAuthenticationToken", "You have to generate or provide a development authentication token in the SpatialOS GDK Editor Settings section to enable connecting to a cloud deployment."));
+			return false;
+		}
+
+		FString DeploymentCheckResult;
+		int32 ExitCode;
+		const USpatialGDKEditorSettings* Settings = GetDefault<USpatialGDKEditorSettings>();
+		bool bIsRunningInChina = GetDefault<USpatialGDKSettings>()->IsRunningInChina();
+		if (!SpatialCommandUtils::HasDevLoginTag(Settings->DevelopmentDeploymentToConnect, bIsRunningInChina, SpatialGDKServicesConstants::SpatialOSDirectory, DeploymentCheckResult, ExitCode))
```

Comment:
Do we need to pass the directory? I.e. would this ever be run from a directory different from `spatial`?